### PR TITLE
Add more v2.x and v3.x fields

### PR DESCRIPTION
**Pull request by aiordache**
Align schema with the unified schema from docker-compose.

Relates to https://github.com/docker/compose/issues/7201

Signed-off-by: aiordache <anca.iordache@docker.com>

### CODE REVIEW

File: schema/compose-spec.json [hunk 1]
```diff
@@ -606,15 +611,26 @@
...
+            },
+            "options": {
+              "type": "object",
+              "additionalProperties": false,
+              "patternProperties": {"^.+$": {"type": "string"}}
             }
           },
-          "additionalProperties": false,
-          "patternProperties": {"^x-": {}}
+          "additionalProperties": false
```
Comment (EricHripko) [inline schema/compose-spec.json:633]: Shouldn't we keep `patternProperties` here or is the removal intentional? This means that no custom/experimental fields can be attached to `ipam` config.

Comment (aiordache) [reply schema/compose-spec.json:633]: My mistake, I'll add it. Thanks for noticing it!

Comment (ndeloof) [reply schema/compose-spec.json]: nice catch!

**Comment by ndeloof** [comment]
@aiordache pushed documentation for missing fields. Please do a quick check to confirm I didn't forgot one/made a bad copy/paste from compose v2 docs :)

**Pull request merged**